engine/src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -56,67 +55,80 @@ class ValidateActor()
     * 2.) Raw inputs and it's coercion
     * 3.) Workflow Options
     * 4.) RuntimeAttributes w.r.t. a backend
+    *
     * @return The result of the validation as instance of `ValidationResult`
     */
-  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = {
-    (for {
-      namespaceWithWorkflow <- Future(NamespaceWithWorkflow.load(wdlSource))
-      validatedInputs <- Future(validateInputs(workflowInputs, namespaceWithWorkflow))
-      validatedWorkflowOptions <- Future(validateWorkflowOptions(workflowOptions, CromwellBackend.backend()))
-      validatedRuntimeAttrs <- Future(validateRuntimeAttributes(namespaceWithWorkflow))
-    } yield ValidationSuccess(namespaceWithWorkflow, validatedInputs, validatedWorkflowOptions, validatedRuntimeAttrs)) recover {
-      case reason => ValidationFailure(reason)
+  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = Future {
```

Comment:
I really dislike having Futures running around inside of an actor as it breaks down the fundamental simplifying abstraction of the actor model - that inside an actor there is a single threaded event loop - making it easier to reason about the internals. 

OTOH we've monstrously overused Futures running around inside of Cromwell's actors, I was hoping this refactor could be the start of a new age where we moved away from that, not adding to it.

One way of handling it would be to use the pipeTo pattern and pipe the Future directly to the caller, although I'm still uneasy with that unless it is a) necessary or b) bubbling all the way back out to a REST endpoint.

What I had meant the other day was literally handling these validations in the single threaded event loop, realizing that given enough load it'd become insufficient. My suggestion was to wait until it's clear that there's a problem to put further structure behind this actor to handle that load via more workers (e.g. your per-request model or francisco's suggestion of a balancing dispatcher) which would be under the supervision of this actor. My primary point there was that we don't know what number of requests would cause problems - it could be 1/s or it could be 100000000000000000000000000/s, and it didn't seem worth adding the structure until we knew it was likely to be a real world problem.

The above said, if that really makes you uneasy I don't care *that* strongly about putting worker structure behind this actor - I've just been saying all along that IMO it was a premature optimization which was trivially fixed down the road.